lib/imagery/sig.c

```diff
@@ -204,7 +208,13 @@ int I_read_signatures(FILE * fd, struct Signature *S)
         return -1;
     }
 
-    while ((n = I_read_one_signature(fd, S)) == 1) ;
+    /* Read marker of original class value presence */
+    if (ver == 2 && fscanf(fd, "%d", &(S->have_oclass)) != 1) {
```

Comment:
Unless the v3 format will be completely different, `ver >= 2` is probably the right code for saying "process something introduced in version 2" like testing for Python 3 behavior with `== 3` versus `>= 3`.

Reply:
Done.